mesonbuild/mlog.py

```diff
@@ -183,7 +184,7 @@ def _log_error(severity, *args, **kwargs):
     elif severity == 'deprecation':
         args = (red('DEPRECATION:'),) + args
     else:
-        assert False, 'Invalid severity ' + severity
+        raise AssertionError('Invalid severity ' + severity)
```

Comment:
I think this should be a `MesonException` or `TypeError`

Reply:
Done